node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -743,30 +678,29 @@ fn validate_lnc_four_forks() -> Result<(), AddUnitError<TestContext>> {
     endorse!(state, rng, g0; ALICE, BOB, CAROL, DAN, GINA, HANNA);
     let f0 = add_unit!(state, rng, FRANK, None; N, N, N, N, F, N, g0, h0; g0)?;
     let d0 = add_unit!(state, rng, DAN, None; N, N, N, N, F, f0, g0, h0)?;
-    let mut d1 = WireUnit {
-        panorama: panorama!(a0, b0, c0, d0, F, f0, g0, h0),
-        creator: DAN,
-        instance_id: 1u64,
-        value: None,
-        seq_number: 0,
-        timestamp: 52.into(),
-        round_exp: 4u8,
-        endorsed: vec![a0, g0].into_iter().collect(), // Endorsements included in the panorama.
-    };
-    assert_eq!(state.validate_lnc(&d1), Some(ERIC));
+    assert_eq!(
+        add_unit!(state, rng, DAN, None; a0, b0, c0, d0, F, f0, g0, h0; a0, g0)
+            .unwrap_err()
+            .cause,
+        UnitError::LncNaiveCitation(ERIC)
+    );
     let mut pre_endorse_state = state.clone();
-    // If we endorse h0, then d1 should be OK since there won't be two pair of equivocating units
+    // If we endorse h0, then d1 should be OK since there won't be two equivocating units
     // cited naively.
     endorse!(state, rng, h0; ALICE, BOB, CAROL, DAN, GINA, HANNA);
-    d1.endorsed.insert(h0);
-    assert_eq!(state.validate_lnc(&d1), None);
+    assert!(add_unit!(state, rng, DAN, None; a0, b0, c0, d0, F, f0, g0, h0; a0, g0, h0).is_ok());
     // It should also work if we had endorsed b0 instead.
     endorse!(pre_endorse_state, rng, b0; ALICE, BOB, CAROL, DAN, GINA, HANNA);
-    d1.endorsed.insert(b0);
-    assert_eq!(pre_endorse_state.validate_lnc(&d1), None);
+    assert!(
+        add_unit!(pre_endorse_state, rng, DAN, None; a0, b0, c0, d0, F, f0, g0, h0; a0, g0, b0)
+            .is_ok()
+    );
     // And it should still work if both were endorsed.
     endorse!(pre_endorse_state, rng, h0; ALICE, BOB, CAROL, DAN, GINA, HANNA);
-    assert_eq!(pre_endorse_state.validate_lnc(&d1), None);
+    assert!(
+        add_unit!(pre_endorse_state, rng, DAN, None; a0, b0, c0, d0, F, f0, g0, h0; a0, g0, b0, h0)
```

Comment:
You never need to `assert!(_.is_ok());`:
```suggestion
        add_unit!(pre_endorse_state, rng, DAN, None; a0, b0, c0, d0, F, f0, g0, h0; a0, g0, b0, h0)?;
```
(Also above.)

Reply:
Obviously.